pipelines/broad/rna_seq/RNAWithUMIsPipeline.wdl

```diff
@@ -0,0 +1,337 @@
+	>>>
+
+	runtime {
+		docker: "us.gcr.io/broad-gatk/gatk:4.2.0.0"
+		disks: "local-disk 100 HDD"
+	}
+
+	output {
+		String sample_name = read_string("sample_name.txt")
+	}
+}
+
+task CopyReadGroupsToHeader {
+	input {
+		File bam_with_readgroups
+		File bam_without_readgroups
+	}
+
+	Int disk_size = ceil(2.0 * size([bam_with_readgroups, bam_without_readgroups], "GB")) + 10
+	String basename = basename(bam_without_readgroups)
+
+	command <<<
+		samtools view -H ~{bam_without_readgroups} > header.sam
+		samtools view -H ~{bam_with_readgroups} | grep "@RG" >> header.sam
+		samtools reheader header.sam ~{bam_without_readgroups} > ~{basename}
+	>>>
+
+	runtime {
+		docker: "us.gcr.io/broad-dsde-methods/samtools@sha256:0e49b0a5d91c203b8c07f5242277c2060b4b8ea54df8b1d123f990a1ad0588b2"
+		disks: "local-disk " + disk_size + " HDD"
+	}
+
+	output {
+		File output_bam = basename
+	}
+}
+
+task CollectRNASeqMetrics {
+	input {
+		File input_bam
+		File input_bam_index
+		String output_bam_prefix
+		File ref_dict
+		File ref_fasta
+		File ref_fasta_index
+		Int preemptible_tries
+		File ref_flat
+		File ribosomal_intervals
+	}
+
+	Float ref_size = size(ref_fasta, "GiB") + size(ref_fasta_index, "GiB") + size(ref_dict, "GiB")
+	Int disk_size = ceil(size(input_bam, "GiB") + ref_size) + 20
+
+	# This jar skips the header check of the ribosomal interval
+	File picard_jar = "gs://broad-dsde-methods-takuto/hydro.gen/picard_ignore_ribosomal_header.jar"
+
+	command {
+		java -Xms5000m -jar ~{picard_jar} CollectRnaSeqMetrics \
+		REF_FLAT=~{ref_flat} \
+		RIBOSOMAL_INTERVALS= ~{ribosomal_intervals} \
+		STRAND_SPECIFICITY=SECOND_READ_TRANSCRIPTION_STRAND \
+		INPUT=~{input_bam} \
+		OUTPUT=~{output_bam_prefix}.rna_metrics
+	}
+
+	runtime {
+		docker: "us.gcr.io/broad-gotc-prod/genomes-in-the-cloud:2.4.3-1564508330"
+		memory: "7 GiB"
+		disks: "local-disk " + disk_size + " HDD"
+		preemptible: preemptible_tries
+	}
+	output {
+		File rna_metrics = output_bam_prefix + ".rna_metrics"
+	}
+}
+
+task CollectMultipleMetrics {
+	input {
+		File input_bam
+		File input_bam_index
+		String output_bam_prefix
+		File ref_dict
+		File ref_fasta
+		File ref_fasta_index
+		Int preemptible_tries
+	}
+
+	Float ref_size = size(ref_fasta, "GiB") + size(ref_fasta_index, "GiB") + size(ref_dict, "GiB")
+	Int disk_size = ceil(size(input_bam, "GiB") + ref_size) + 20
+
+	File ref_flat = "gs://gcp-public-data--broad-references/hg38/v0/GRCh38_gencode.v27.refFlat.txt"
+
+	command {
+		java -Xms5000m -jar /usr/gitc/picard.jar CollectMultipleMetrics \
+		INPUT=~{input_bam} \
+		OUTPUT=~{output_bam_prefix} \
+		PROGRAM=CollectInsertSizeMetrics \
+		PROGRAM=CollectAlignmentSummaryMetrics \
+		REFERENCE_SEQUENCE=~{ref_fasta}
+
+		ls > ls.txt
+	}
+
+	runtime {
+		docker: "us.gcr.io/broad-gotc-prod/genomes-in-the-cloud:2.4.3-1564508330"
+		memory: "7 GiB"
+		disks: "local-disk " + disk_size + " HDD"
+		preemptible: preemptible_tries
+	}
+
+	output {
+		File ls = "ls.txt"
```

Comment:
I'm assuming this is just a place holder and that you intend to return all of the metrics files?